cookbooks/aws-parallelcluster-platform/templates/nvidia-imex/nvidia-imex.service.erb

```diff
@@ -0,0 +1,26 @@
+[Unit]
+Description=NVIDIA IMEX service
+After=network-online.target
+Requires=network-online.target
+
+[Service]
+Environment="KRB5_CLIENT_KTNAME=/etc/krb5.keytab"
+User=root
+PrivateTmp=false
+Type=forking
+TimeoutStartSec=infinity
+
+ExecStart=/usr/bin/nvidia-imex -c <%= node['cluster']['nvidia']['imex']['shared_dir'] %>/config.cfg
+
+LimitCORE=infinity
+
+Restart=on-failure
```

Comment:
Why restarting on failure rather than restarting always?
One of the advantage of restarting always is that we capture whatever termination scenario

Reply:
On those instance types where it should not run, the service is disabled, so the restart policy is not even taken into account.
The advantage of always compared to on-failure is that the service gets restarted regardless the exit code. This is just a safety net. 

However since you took the unit definition from NVIDIA recommedation, they probably have good reasons to prefer on-failure rather than always.
Let's go with on-failure.